File: frontend/src/dto/user.dto.ts
import { UUID } from "crypto";

export interface UserDTO {
    userId: UUID;
    email: string;
    username: string;
    phone: string;
    role: RoleDTO;
    status: boolean;
}

export interface RoleDTO {
    name: string;
    description: string;
}

export enum ROLE_NAME {
    ADMIN = "ADMIN",
    BRANCH_MANAGER = "BRANCH_MANAGER",
    RESTAURANT_OWNER = "RESTAURANT_OWNER",
    WAITER = "WAITER",
    RECEPTIONIST = "RECEPTIONIST"
}

export interface SignupRequest {
    email: string;
    password: string;
    username: string;
    phone: string;
}

File: backend/src/main/java/com/example/backend/dto/UserDTO.java
package com.example.backend.dto;

import java.util.UUID;

public class UserDTO {
    
    private UUID userId;
    private String email;
    private String username;
    private String phone;
    private RoleDTO role;
    private boolean status;
    
    public UUID getUserId() {
        return userId;
    }
    public void setUserId(UUID userId) {
        this.userId = userId;
    }
    public String getEmail() {
        return email;
    }
    public void setEmail(String email) {
        this.email = email;
    }
    public String getUsername() {
        return username;
    }
    public void setUsername(String username) {
        this.username = username;
    }
    public String getPhone() {
        return phone;
    }
    public void setPhone(String phone) {
        this.phone = phone;
    }
    public RoleDTO getRole() {
        return role;
    }
    public void setRole(RoleDTO role) {
        this.role = role;
    }
    public boolean isStatus() {
        return status;
    }
    public void setStatus(boolean status) {
        this.status = status;
    }
    
}
